packages/desktop-client/src/components/reports/reports/Spending.tsx

```diff
@@ -198,30 +212,45 @@ export function Spending() {
               paddingRight: 5,
             }}
           >
-            Compare
+            <Trans>Compare</Trans>
           </Text>
           <Select
-            value={compare}
-            onChange={e => {
-              setCompare(e);
+            value={
+              mode !== 'sliding-window' || reportMode === 'single-month'
+                ? start
+                : 'current-month'
+            }
+            onChange={newStart => {
+              if (newStart === 'current-month') {
+                setMode('sliding-window');
+                setStart(monthUtils.currentMonth());
+                return;
+              }
+              setMode('static');
+              setStart(newStart);
             }}
-            options={allIntervals.map(({ name, pretty }) => [name, pretty])}
+            options={[
+              ...(reportMode === 'single-month'
+                ? []
+                : [['current-month', t('Current Month')] as const]),
```

Comment:
I'd suggest we remove the actual current month from the list so it's not duplicated. Could we also offer a "last month" option in the "to" selection and create a "live" graph when "single month" button is chosen?

Reply:
I actually removed the notion of "current month" and instead aligned the functionality with the other reports to have a more consistent interface. It did seem like a good idea in the beginning to use the "current month" label, but it actually has a fair few complexities that I hadn't considered. Maybe we will bring it back in a future revision. But for now - no.

--

Added the "live/static" button.